product_pricelist_brand/models/product_pricelist.py

```diff
@@ -0,0 +1,140 @@
+                    else:
+                        # complete formula
+                        price_limit = price
+                        price = (
+                            price - (price * (rule.price_discount / 100))) or 0.0
+                        if rule.price_round:
+                            price = tools.float_round(
+                                price, precision_rounding=rule.price_round)
+
+                        if rule.price_surcharge:
+                            price_surcharge = convert_to_price_uom(
+                                rule.price_surcharge)
+                            price += price_surcharge
+
+                        if rule.price_min_margin:
+                            price_min_margin = convert_to_price_uom(
+                                rule.price_min_margin)
+                            price = max(price, price_limit + price_min_margin)
+
+                        if rule.price_max_margin:
+                            price_max_margin = convert_to_price_uom(
+                                rule.price_max_margin)
+                            price = min(price, price_limit + price_max_margin)
+                    suitable_rule = rule
+                # Final price conversion into pricelist currency
+                if suitable_rule and price and \
+                        suitable_rule.compute_price != 'fixed' and \
+                        suitable_rule.base != 'pricelist':
+                    if suitable_rule.base == 'standard_price':
+                        cur = product.cost_currency_id
+                    else:
+                        cur = product.currency_id
+                    price = cur._convert(price, self.currency_id,
+                                         self.env.user.company_id,
+                                         date, round=False)
+
+                if price:
+                    result[product.id] = (
+                        price, suitable_rule and suitable_rule.id or False)
+        return result
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    applied_on = fields.Selection(selection_add=[
+        ("2_product_brand", "Product Brand")])
+    brand_id = fields.Many2one(
+        'product.brand', 'Product Brand', ondelete='cascade',
+        help="Specify a product brand if this rule only applies to products "
+             "belonging to this brand. Keep empty otherwise.")
+
+    @api.multi
+    @api.depends('brand_id', 'categ_id', 'product_tmpl_id', 'product_id',
+                 'compute_price', 'fixed_price', 'pricelist_id',
+                 'percent_price', 'price_discount', 'price_surcharge')
+    def _get_pricelist_item_name_price(self):
+        super(ProductPricelistItem, self)._get_pricelist_item_name_price()
+        for item in self:
+            if item.brand_id:
+                item.name = _("Brand: %s") % (item.brand_id.name)
+
+    @api.onchange('applied_on')
+    def _onchange_applied_on(self):
```

Comment:
In the base module there is no `_onchange_applied_on` but updates in create and write, it is more robust to stick to that.
Plus don't forget the method `_check_product_consistency`.